src/test/java/org/opentripplanner/middleware/persistence/PersistenceTest.java

```diff
@@ -42,13 +60,161 @@ public void canDeleteUser() {
+        // tidy up
+        Persistence.tripRequests.removeById(tripRequest.id);
+    }
+
+    @Test
+    public void canDeleteTripRequest() {
+        String userId = "123456";
+        TripRequest tripRequestToDelete = createTripRequest(userId);
+        Persistence.tripRequests.removeById(tripRequestToDelete.id);
+        TripRequest tripRequest = Persistence.tripRequests.getById(tripRequestToDelete.id);
+        assertNull(tripRequest, "Deleted TripRequest should no longer exist in database (should return as null).");
+    }
+
+    @Test
+    public void canCreateTripSummaryWithError() {
+        TripSummary tripSummary = createTripSummaryWithError();
+        TripSummary retrieved = Persistence.tripSummaries.getById(tripSummary.id);
+        System.out.println("Retrieved trip summary with error:" + retrieved.toString());
+        assertEquals(tripSummary.id, retrieved.id, "Found Trip summary ID should equal inserted ID.");
+        // tidy up
+        Persistence.tripSummaries.removeById(tripSummary.id);
+    }
+
+    @Test
+    public void canCreateTripSummary() {
+        TripSummary tripSummary = createTripSummary();
+        TripSummary retrieved = Persistence.tripSummaries.getById(tripSummary.id);
+        System.out.println("Retrieved trip summary:" + retrieved.toString());
+        assertEquals(tripSummary.id, retrieved.id, "Found Trip summary ID should equal inserted ID.");
+        // tidy up
+        Persistence.tripSummaries.removeById(tripSummary.id);
+    }
+
+    @Test
+    public void canDeleteTripSummary() {
+        TripSummary tripSummaryToDelete = createTripSummary();
+        Persistence.tripSummaries.removeById(tripSummaryToDelete.id);
+        TripSummary tripSummary = Persistence.tripSummaries.getById(tripSummaryToDelete.id);
+        assertNull(tripSummary, "Deleted trip summary should no longer exist in database (should return as null).");
+    }
+
+    @Test
+    public void canGetFilteredTripRequestsWithFromAndToDate() {
+        int limit = 3;
+        String TRIP_REQUEST_DATE_CREATED_FIELD_NAME = "dateCreated";
+        String TRIP_REQUEST_USER_ID_FIELD_NAME = "userId";
+
+        OtpUser user = createUser(TEST_EMAIL);
+
+        List<TripRequest> tripRequests = createTripRequests(limit, user.id);
+
+        LocalDateTime fromStartOfDay = LocalDate.now().atTime(LocalTime.MIN);
+        LocalDateTime toEndOfDay = LocalDate.now().atTime(LocalTime.MAX);
+
+        Bson filter = Filters.and(gte(TRIP_REQUEST_DATE_CREATED_FIELD_NAME, Date.from(fromStartOfDay.atZone(ZoneId.systemDefault()).toInstant())),
+            lte(TRIP_REQUEST_DATE_CREATED_FIELD_NAME, Date.from(toEndOfDay.atZone(ZoneId.systemDefault()).toInstant())),
+            eq(TRIP_REQUEST_USER_ID_FIELD_NAME, user.id));
+
+        List<TripRequest> result = Persistence.tripRequests.getFilteredWithLimit(filter, limit);
+        assertEquals(result.size(), tripRequests.size());
+
+        // tidy up
+        deleteTripRequests(tripRequests);
```

Comment:
Is there a way to put this in an afterEach method that clears tripRequests after each test case?

Reply:
Moved the trip history tests into a separate class and made use of after each.